syntax = "proto3";

package bank;

service Bank{
    // Unary
	rpc BranchQuery (CustomerQueryRequest) returns (CustomerQueryResponse);
    // Unary
	rpc BranchWithdraw (CustomerTransactRequest) returns (NotificaionResponse);
    // Unary
	rpc BranchDeposit (CustomerTransactRequest) returns (NotificaionResponse);
    // Unary
	rpc BranchPropogateWithdraw (CustomerTransactRequest) returns (NotificaionResponse);
    // Unary
	rpc BranchPropogateDeposit (CustomerTransactRequest) returns (NotificaionResponse);
}

message CustomerQueryRequest{
    int32 id = 1;
}

message CustomerTransactRequest{
    int32 id = 1;
    int32 money = 2;
}

message NotificaionResponse{
    string msg = 1;
}

message CustomerQueryResponse{
    int32 msg = 1;
}